fcrepo-auth-webac/src/test/java/org/fcrepo/integration/auth/webac/WebACRecipesIT.java

```diff
@@ -1325,4 +1332,596 @@ public void testCreateAclWithAccessToClassForBinary() throws Exception {
+        assertEquals(HttpStatus.SC_FORBIDDEN, getStatus(putChild));
+
+        // Put the child as Admin
+        setAuth(putChild, "fedoraAdmin");
+        assertEquals(HttpStatus.SC_CREATED, getStatus(putChild));
+
+        // Try to delete the child as user
+        final HttpDelete deleteChild = new HttpDelete(indirectUri + "/" + id);
+        setAuth(deleteChild, username);
+        assertEquals(HttpStatus.SC_FORBIDDEN, getStatus(deleteChild));
+
+        // Try to delete the indirect container
+        final HttpDelete deleteIndirect = new HttpDelete(indirectUri);
+        setAuth(deleteIndirect, username);
+        assertEquals(HttpStatus.SC_FORBIDDEN, getStatus(deleteIndirect));
+
+        // Ensure we can still write to the writeable resource.
+        testCanWrite(writeableResource, username);
+
+    }
+
+    @Test
+    public void testIndirectRelationshipOK() throws IOException {
+        final String targetResource = "/rest/" + getRandomUniqueId();
+        final String writeableResource = "/rest/" + getRandomUniqueId();
+        final String username = "user28";
+
+        final String targetUri = ingestObj(targetResource);
+
+        final String readonlyString = "@prefix acl: <http://www.w3.org/ns/auth/acl#> .\n" +
+                "<#readauthz> a acl:Authorization ;\n" +
+                "   acl:agent \"" + username + "\" ;\n" +
+                "   acl:mode acl:Read, acl:Write ;\n" +
+                "   acl:accessTo <" + targetResource + "> .";
+        ingestAclString(targetUri, readonlyString, "fedoraAdmin");
+
+        // User can read target resource.
+        final HttpGet get1 = getObjMethod(targetResource);
+        setAuth(get1, username);
+        assertEquals(HttpStatus.SC_OK, getStatus(get1));
+
+        // User can patch target resource.
+        final String patch = "INSERT DATA { <> <http://purl.org/dc/elements/1.1/title> \"Changed it\"}";
+        final HttpEntity patchEntity = new StringEntity(patch, sparqlContentType);
+        try (final CloseableHttpResponse resp = (CloseableHttpResponse) PATCH(targetUri, patchEntity,
+                username)) {
+            assertEquals(HttpStatus.SC_NO_CONTENT, getStatus(resp));
+        }
+
+        // Make a user writable container.
+        final String writeableUri = ingestObj(writeableResource);
+        final String writeableAcl = "@prefix acl: <http://www.w3.org/ns/auth/acl#> .\n" +
+                "<#writeauth> a acl:Authorization ;\n" +
+                "   acl:agent \"" + username + "\" ;\n" +
+                "   acl:mode acl:Read, acl:Write ;\n" +
+                "   acl:accessTo <" + writeableResource + "> ;\n" +
+                "   acl:default <" + writeableResource + "> .";
+        ingestAclString(writeableUri, writeableAcl, "fedoraAdmin");
+
+        // Ensure we can write to the writeable resource.
+        testCanWrite(writeableResource, username);
+
+        // Try to create indirect container referencing readonly resource with POST.
```

Comment:
Locally, I changed the ACL above to make `targetResource` actually read-only... this test now fails.